packages/amplication-server/src/core/build/build.service.spec.ts

```diff
@@ -468,4 +547,67 @@ describe('BuildService', () => {
     );
     expect(actionServiceLogMock).toBeCalledTimes(0);
   });
+
+  test('should catch an error when trying to build', async () => {
+    const EXAMPLE_ERROR = new Error('ExampleError');
+    const logArgs = {
+      step: EXAMPLE_ACTION_STEP,
+      enumError: EnumActionLogLevel.Error,
+      error: EXAMPLE_ERROR
+    };
+    const completeArgs = {
+      step: EXAMPLE_ACTION_STEP,
+      enumStatus: EnumActionStepStatus.Failed
+    };
+    const failStatus = EnumBuildStatus.Failed;
+    const catchUpdateArgs = {
+      where: { id: EXAMPLE_BUILD_ID },
+      data: { status: failStatus }
+    };
+    // eslint-disable-next-line
+    // @ts-ignore
+    winston.createLogger.mockImplementation(() => MOCK_LOGGER);
+    // eslint-disable-next-line
+    // @ts-ignore
+    winston.transports.Console = jest.fn(() => MOCK_CONSOLE_TRANSPORT);
+    // eslint-disable-next-line
+    // @ts-ignore
+    DataServiceGenerator.createDataService.mockImplementation(() => {
+      throw EXAMPLE_ERROR;
+    });
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    loggerChildErrorMock.mockImplementation((error: Error) => {
+      return;
+    });
+    const buildId = EXAMPLE_BUILD_ID;
+    expect(await service.build(buildId)).toBeUndefined();
+    expect(findOneMock).toBeCalledTimes(1);
+    expect(findOneMock).toBeCalledWith({
+      where: { id: buildId }
+    });
+    expect(loggerChildMock).toBeCalledTimes(1);
+    expect(loggerChildMock).toBeCalledWith({
+      buildId: buildId
+    });
+    expect(loggerChildInfoMock).toBeCalledTimes(2);
+    expect(loggerChildInfoMock.mock.calls).toEqual([
+      [JOB_STARTED_LOG],
+      [JOB_DONE_LOG]
+    ]);
+    expect(loggerChildErrorMock).toBeCalledTimes(1);
+    expect(loggerChildErrorMock).toBeCalledWith(EXAMPLE_ERROR);
+    expect(actionServiceLogMock).toBeCalledTimes(1);
+    expect(actionServiceLogMock).toBeCalledWith(
+      logArgs.step,
+      logArgs.enumError,
+      logArgs.error
+    );
+    expect(actionServiceCompleteMock).toBeCalledTimes(1);
+    expect(actionServiceCompleteMock).toBeCalledWith(
+      completeArgs.step,
+      completeArgs.enumStatus
+    );
+    expect(updateMock).toBeCalledTimes(2);
+    expect(updateMock).toBeCalledWith(catchUpdateArgs);
```

Comment:
If `updateMock` is expected to be called twice assert on `updateMock.mock.calls` instead.